core/src/main/java/io/grpc/internal/KeepAliveManager.java

```diff
@@ -86,7 +86,10 @@ public void run() {
               TimeUnit.NANOSECONDS);
         } else if (state == State.PING_DELAYED) {
           // We have received some data. Reschedule the ping with the new time.
-          pingFuture = scheduler.schedule(sendPing, nextKeepaliveTime - ticker.read(),
+          pingFuture = scheduler.schedule(
+              sendPing,
+              // normalized as some Netty executor service does not accept a negative delay
```

Comment:
Can you file a bug? It is [part of the interface](http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html):
> Zero and negative delays (but not periods) are also allowed in schedule methods